onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -1799,7 +1816,11 @@ class PlannerImpl {
         auto* node = graph_viewer_.GetNode(node_index);
         for (auto it = node->OutputNodesBegin(); it != node->OutputNodesEnd(); ++it) {
           // if the output node is not in the same stream, generate a trigger point
-          if (node_stream_map_[it->Index()] != i) {
+          if (node_stream_map_[it->Index()] != i
+#ifdef ENABLE_TRAINING
```

Comment:
do you need training flag guard here? it should also works for inference, right? although there is no case that inference mode using yield op.

Reply:
btw, please add comments that why we don't insert trigger step here.

Reply:
added comments. As we discussed offline, I'd like to keep this logic transparent to inference scenario